ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -484,12 +483,16 @@ forkBlockProduction IS{..} =
   New network layer
 -------------------------------------------------------------------------------}
 
+data SharedState m s a = SharedState (m s) (s -> a)
```

Comment:
I still argue that this whole thing should not be necessary. The fix is in the wrong place. The block fetch and chain sync client _should not_ require an ordering on their startup. @dcoutts @coot